doc/CMakeLists.txt

```diff
@@ -28,49 +24,145 @@ if (BUILD_DOCS)
+            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile_xml ${DOXYGEN_SOURCE_FILES}
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile_xml
+            COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/doxygen_xml.stamp
+    )
+
+    add_custom_target(doxygen_xml
+            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/doxygen_xml.stamp
+    )
+
+    # Doxygen HTML outputs
+    string(JOIN "\n" DOXYFILE_HTML_CONTENTS
+            "${DOXYFILE_CONTENTS}"
+            "FAIL_ON_WARNINGS=YES"
+            "WARN_AS_ERROR=FAIL_ON_WARNINGS_PRINT"
+            "HTML_OUTPUT=${SPHINX_BUILD_DIR}/html_extra/doxygen"
+            "INLINE_INHERITED_MEMB=YES"
+    )
+    file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile_html ${DOXYFILE_HTML_CONTENTS})
+    add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/doxygen_html.stamp
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile_html ${DOXYGEN_SOURCE_FILES}
+            COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/Doxyfile_html
+            COMMAND ${CMAKE_COMMAND} -E touch ${CMAKE_CURRENT_BINARY_DIR}/doxygen_html.stamp
+    )
+    add_custom_target(doxygen_html
+            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/doxygen_html.stamp
+    )
+
+    ####################################################################################################
+    # Sphinx outputs
+    ####################################################################################################
+
+    # Sphinx HTML documentation
+    add_custom_command(
+            OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/html.stamp"
+            DEPENDS doxygen_xml python_binding
+                    ${SPHINX_SOURCE_FILES}
+                    COMMAND ${CMAKE_COMMAND} -E env ${PYTHON_RUN_ENV} BUILDDIR=${SPHINX_BUILD_DIR}
+                    ${SPHINX_BUILD} -M html
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/source
+                                    ${CMAKE_CURRENT_BINARY_DIR}/build
+                                    ${SPHINX_BUILD_OPTS}
+            COMMAND ${CMAKE_COMMAND} -E touch "${CMAKE_CURRENT_BINARY_DIR}/html.stamp"
+    )
+
+    add_custom_target(html
+            DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/html.stamp
+    )
+
+    # Sphinx PDF documentation
+    add_custom_command(
+            OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/build/gdal.pdf
+            DEPENDS ${SPHINX_SOURCE_FILES}
+            COMMAND ${CMAKE_COMMAND} -E env BUILDDIR=${SPHINX_BUILD_DIR}
+                    ${SPHINX_BUILD} -M latexpdf
+                                    ${CMAKE_CURRENT_SOURCE_DIR}/source
+                                    ${CMAKE_CURRENT_BINARY_DIR}/build
+                                    ${SPHINX_BUILD_OPTS}
+    )
+
+    add_custom_target(latexpdf DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/build/gdal.pdf)
+
+    # Sphinx manpage documentation
```

Comment:
mkgdaldist.sh will need to be updated because it does a (cd doc; make man)

Reply:
I could replace `make man` with a direct `sphinx-build` invocation, or something like 

```
mkdir build-man
cd build-man
cmake ..
cmake --build . --target man
```

The `man` target does not currently depend on GDAL. But building docs for some utilities (like `gdalinfo`) requires executing the utility and collecting the output. So we either need to build GDAL to generate the man pages, or we need to be satisfied with the output of whatever GDAL version is found in the current path. Any thoughts on this?

Reply:
> The `man` target does not currently depend on GDAL. But building docs for some utilities (like `gdalinfo`) requires executing the utility and collecting the output. So we either need to build GDAL to generate the man pages, or we need to be satisfied with the output of whatever GDAL version is found in the current path. Any thoughts on this?

we probably need to do a GDAL build of the version we release :-)   A minimal build with -DGDAL_BUILD_OPTIONAL_DRIVERS=OFF -DOGR_BUILD_OPTIONAL_DRIVERS=OFF should do it

Reply:
Added in [1806ba8](https://github.com/OSGeo/gdal/pull/11951/commits/1806ba8e1f17669d585f250efe11282f14a48743), and then changed in [8456bb6](https://github.com/OSGeo/gdal/pull/11951/commits/8456bb6fda84e5b16372bcc874cd285bdd643738) to avoid building GDAL once for the man pages and again for the doc snapshot.

Reply:
@dbaston Did you try running the update ./mkgdaldist.sh script ?

Reply:
Yes, via `./mkgdaldist.sh 3110 -branch cmake-build-docs -url https://github.com/dbaston/gdal`